### PR TITLE
Refine IPEX integration for accelerate

### PR DESCRIPTION
**Pull request by jianan-gu**

### CODE REVIEW

**Comment by jianan-gu** [comment]
@pacman100 Hi could you have a review on the jit refinement (changes refer to https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1299)? thanks!   
cc @sywangyi

**Comment by sywangyi** [comment]
Hi. @pacman100 could you help review and merge the change to your ipex accelerate PR? thanks

File: src/accelerate/utils/dataclasses.py [hunk 1]
```diff
@@ -266,25 +266,37 @@ def torch_jit_model_eval(self, model, dataloader, training=False):
...
+                        for key in example_batch:
+                            example_tensor = torch.ones_like(example_batch[key])
+                            jit_inputs.append(example_tensor)
+                        jit_inputs = tuple(jit_inputs)
                         jit_model = torch.jit.trace(jit_model, jit_inputs, strict=False)
-                        jit_model = torch.jit.freeze(jit_model)
-                jit_model(**example_batch)
+                jit_model = torch.jit.freeze(jit_model)
+                with torch.no_grad():
+                    jit_model(**example_batch)
```
Comment (pacman100) [inline src/accelerate/utils/dataclasses.py:291]: this is repeating twice, wanted to make sure if this is intended?

**Review by pacman100 — COMMENTED** [review]
Thank you @jianan-gu for iterating and refinement 🤗! Left a comment

File: src/accelerate/utils/dataclasses.py (hunk 1 above)
Comment (jianan-gu) [reply src/accelerate/utils/dataclasses.py:291]: yes, it is intended to make sure there is no unexpected error outside of the `try`.

**Review by pacman100 — APPROVED** [review]

**Pull request merged**